### PR TITLE
metrics: add reporting of `closed request prematurely`

### PR DESCRIPTION
**Pull request by mwitkow**
Adds monitoring of the `closed request prematurely` problem on the proxy.
Fixes not reporting the incoming request (must have been dropped in merge resolutions).
https://github.com/coreos/etcd/issues/3405

### CODE REVIEW

**Comment by mwitkow-io** [comment]
@xiang90 adding this monitoring will help us track if https://github.com/coreos/etcd/issues/3405 ever comes back. We did see it running 2.1.1.

**Comment by xiang90** [comment]
@mwitkow-io After upgrading etcd, the issue is fixed?

**Comment by mwitkow-io** [comment]
@xiang90 not sure, we think it may. That's precisely why I want to add this monitoring, so I don't need to scavange hundred of machines for logs :)

Plus, it's for completeness of the the monitoring variable.

File: proxy/reverse.go [hunk 1]
```diff
@@ -67,6 +67,8 @@ func (p *reverseProxy) ServeHTTP(rw http.ResponseWriter, clientreq *http.Request
 	*proxyreq = *clientreq
 	startTime := time.Now()
 
+	reportIncomingRequest(clientreq)
```
Comment (xiang90) [inline proxy/reverse.go]: can we separate this out into another commit at least?

**Comment by xiang90** [comment]
LGTM

**Comment by mwitkow-io** [comment]
Separated both fixes into two commits.

File: proxy/reverse.go [hunk 2]
```diff
@@ -67,6 +67,8 @@ func (p *reverseProxy) ServeHTTP(rw http.ResponseWriter, clientreq *http.Request
 	*proxyreq = *clientreq
 	startTime := time.Now()
 
+  reportIncomingRequest(clientreq)
```
Comment (xiang90) [inline proxy/reverse.go]: gofmt the code?

**Comment by simonvanderveldt** [comment]
Can this PR be merged?
(it might need a rebase)

We're seeing a lot of `proxy: client 127.0.0.1:46282 closed request prematurely` messages and would like to monitor this.
The issue mentioned in #3405 doesn't seem of relevance since we don't run anything against etcd apart from fleet. Is there a way to find out who the client is that closed the request?

**Comment by spzala** [comment]
@xiang90 @mwitkow hello, changes related to one of the two commits are already added as part of some other PR. Adding the metric mentioned here seems still makes sense. I am closing this one and have created a clean PR. Thanks!
